### ISSUE TITLE
Task: Language Identification

### ISSUE THREAD

**Issue by ibevers** [sensein/senselab#85]
### Description

As far as I understand, this should take in an Audio or a ScriptLine and output a Language object

#### Audio
- [x] Create a general API
- [x] Select and implement a default model
- [x] Get working with the default model
- [x] Tutorial
- [x] Documentation

#### Text
- [ ] Create a general API
- [ ] Select and implement a default model
- [ ] Get working with the default model
- [ ] Tutorial
- [ ] Documentation

### Freeform Notes

Might want to have examples that cover a wide range of languages, or we could just trust the model developer. Ideally, we should have multi-class output, so if a given input includes more than one language, the output will reflect that.

**Comment by fabiocat93** [comment]
correct. we may want to have 2 modules, one in `senselab.text` and one in `senselab.audio`

**Comment by fabiocat93** [comment]
For now, I have implemented audio-based language identification using Speechbrain's models. These models work assuming that only one language is included in a clip. In the future, we may want to integrate the Whisper model. This should be easy to implement since we already use the same model for speech-to-text and allow the identification of multiple languages in the same clip. Here is a first draft of how Whisper works:

```python
import torch
from transformers import AutoModelForSpeechSeq2Seq, AutoProcessor, pipeline
from datasets import load_dataset


device = "cuda:0" if torch.cuda.is_available() else "cpu"
torch_dtype = torch.float16 if torch.cuda.is_available() else torch.float32

model_id = "openai/whisper-tiny"

model = AutoModelForSpeechSeq2Seq.from_pretrained(
    model_id, torch_dtype=torch_dtype, low_cpu_mem_usage=True
)
model.to(device)

processor = AutoProcessor.from_pretrained(model_id)

pipe = pipeline(
    "automatic-speech-recognition",
    model=model,
    tokenizer=processor.tokenizer,
    feature_extractor=processor.feature_extractor,
    chunk_length_s=30,
    batch_size=16,
    torch_dtype=torch_dtype,
    device=device,
    return_language=True
)

dataset = load_dataset("distil-whisper/librispeech_long", "clean", split="validation")
sample = dataset[0]["audio"]

result = pipe(sample)
result['chunks'][0]['language']
```

**Comment by fabiocat93** [comment]
@ibevers I have implemented the speech-based version of this. Can you do the same with text-based language identification? Simply integrating huggingface models for this should be more than fine for now (https://huggingface.co/models?search=language%20detection)

**Comment by ibevers** [comment]
@fabiocat93 Thank you for your patience. I can take a look at this after next week

**Comment by fabiocat93** [comment]
> [@fabiocat93](https://github.com/fabiocat93) Thank you for your patience. I can take a look at this after next week

I will extend your deadline one more time (from Dec 16 to Jan 14). Please, let me know if you face any blockers
